From 1061fd730e6c3519497551f72def06d3379d9b57 Mon Sep 17 00:00:00 2001
From: Naresh Solanki <naresh.solanki@9elements.com>
Date: Tue, 10 Mar 2026 13:55:44 +0000
Subject: [PATCH 4/4] hwmontempsensor: Add SPYRE chip

Upstream-Status: Pending

Signed-off-by: Naresh Solanki <naresh.solanki@9elements.com>
---
 src/hwmon-temp/HwmonTempMain.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/hwmon-temp/HwmonTempMain.cpp b/src/hwmon-temp/HwmonTempMain.cpp
index ffc135d23..1735b172a 100644
--- a/src/hwmon-temp/HwmonTempMain.cpp
+++ b/src/hwmon-temp/HwmonTempMain.cpp
@@ -86,6 +86,7 @@ static const I2CDeviceTypeMap sensorTypes{
     {"PT5161L", I2CDeviceType{"pt5161l", true}},
     {"SBTSI", I2CDeviceType{"sbtsi", true}},
     {"SI7020", I2CDeviceType{"si7020", false}},
+    {"SPYRE", I2CDeviceType{"spyre", true}},
     {"TMP100", I2CDeviceType{"tmp100", true}},
     {"TMP1075", I2CDeviceType{"tmp1075", true}},
     {"TMP112", I2CDeviceType{"tmp112", true}},
-- 
2.43.0

